include/light_export.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 woidpointer@gmail.com
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef INCLUDE_EXPORT_H_
+#define INCLUDE_EXPORT_H_
+
+/* =====   LIGHT_API : control library symbols visibility   ===== */
+#ifndef LIGHT_VISIBILITY
+#if defined(__GNUC__) && (__GNUC__ >= 4)
+#define LIGHT_VISIBILITY __attribute__((visibility("default")))
+#else
+#define LIGHT_VISIBILITY
+#endif
+#endif
+
+/* API call convention*/
+#if WIN32
+#define LIGHT_API_CALL __stdcall
```

Comment:
Any reason for choosing `__stdcall` over `__cdecl` ?

Reply:
Not really, I usually use this convention. Shall I change it? Do you have any positive or negative experience?

Reply:
To be honest I have not faced issues with either of them, but checking on SO https://stackoverflow.com/a/6635067/1438522 made me think about it a bit, considering the answer there, it may be a safer alternative to go with the standard cdecl.